File: apps/web/android-src/MainActivity.java
package com.fitech.app;

import android.os.Bundle;
import android.view.KeyEvent;
import com.getcapacitor.BridgeActivity;

public class MainActivity extends BridgeActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // Must be before super.onCreate: BridgeActivity.onCreate calls load()
        // which builds the Bridge from initialPlugins. After that it's too late.
        registerPlugin(MediaButtonPlugin.class);
        registerPlugin(FiTechTTSPlugin.class);
        super.onCreate(savedInstanceState);
    }

    // Backup handler for wired earbuds that send direct KeyEvents to the activity
    @Override
    public boolean dispatchKeyEvent(KeyEvent event) {
        if (event.getAction() == KeyEvent.ACTION_DOWN) {
            MediaButtonPlugin plugin = MediaButtonPlugin.instance;
            if (plugin != null && plugin.handleTap(event.getKeyCode())) {
                return true;
            }
        }
        return super.dispatchKeyEvent(event);
    }
}

File: apps/web/android-src/MainActivity.kt
package com.fitech.app

import android.view.KeyEvent
import com.getcapacitor.BridgeActivity

class MainActivity : BridgeActivity() {

    override fun onCreate(savedInstanceState: android.os.Bundle?) {
        // Capacitor's plugin registry is initialized inside BridgeActivity.onCreate,
        // so registration must happen after super.onCreate (otherwise the plugin
        // is silently dropped and JS never receives tap events).
        super.onCreate(savedInstanceState)
        registerPlugin(MediaButtonPlugin::class.java)
    }

    // Backup handler for wired earbuds that send direct KeyEvents to the activity
    override fun dispatchKeyEvent(event: KeyEvent): Boolean {
        if (event.action == KeyEvent.ACTION_DOWN) {
            val plugin = MediaButtonPlugin.instance
            if (plugin != null && plugin.handleTap(event.keyCode)) {
                return true
            }
        }
        return super.dispatchKeyEvent(event)
    }
}
